sycl/source/detail/graph_impl.hpp

```diff
@@ -473,6 +479,15 @@ class node_impl : public std::enable_shared_from_this<node_impl> {
     }
   }
 
+  /// Returns true if this node should be enqueued to the backend, if not only
+  /// its dependencies are considered.
+  bool requiresEnqueue() const {
+    return MNodeType != node_type::empty &&
+           MNodeType != node_type::ext_oneapi_barrier &&
+           MNodeType != node_type::async_malloc &&
+           MNodeType != node_type::async_free;
```

Comment:
Not 100% about this, but a switch on `MNodeType` might better. With all fall through of all the types to return true on, then default to false for everything else.

Reply:
Good shout, I think that is a little cleaner yeah.